esp/esp/themes/theme_data/fruitsalad/templates/main.html

```diff
@@ -209,17 +209,29 @@ <h1 class="title">{% block content_title %}&nbsp;{% endblock %}</h1>
 {% endblock %}
 
       <div id="content">
+          {% if request.show_perm_info %}
+          <div class="alert alert-info">
+              <span class="glyphicon glyphicon-info-sign" aria-hidden="true"></span>
+              Permission/deadline controlling this page: {{ request.perm_names|join:", " }}
+              <br />
+              {% if request.roles_with_perm %}
+                  Roles with permission to view this page: {{ request.roles_with_perm|join:", " }}
+              {% else %}
+                  No roles with permission to view this page!
```

Comment:
super nit: I'd say "no roles have permission..." since that is clearer if you don't know what it might say if there were such roles.